Recent CI runs failing with NetCDF: HDF error

**Description**

Recent CI (and local) runs of our tests fail with messages like

```
E   RuntimeError: NetCDF: HDF error                                                                                   
                                                                                                                      
netCDF4/_netCDF4.pyx:1887: RuntimeError                                                                               
                                                                                                                      
During handling of the above exception, another exception occurred:                                                   
                                                                                                                      
self = <heat.core.tests.test_io.TestIO testMethod=test_save_netcdf>                                                   
                                                                                                                      
    def test_save_netcdf(self):                                                                                       
        # netcdf support is optional                                                                                  
        if not ht.io.supports_netcdf():                                                                               
            return                                                                                                    
                                                                                                                      
        # local unsplit data                                                                                          
        local_data = ht.arange(100)                                                                                   
>       ht.save_netcdf(local_data, self.NETCDF_OUT_PATH, self.NETCDF_VARIABLE)                                        
                                                                                                                      
heat/core/tests/test_io.py:373:   
```

**To Reproduce**
Steps to reproduce the behavior:
1. Which module/class/function is affected?
heat/core/tests/test_io.py
2. What are the circumstances under which the bug appears?
ANY, just run from current master
3. What is the exact error-message/errorous behavious?
cf. above.


**Expected behavior**
Tests should run successfully.

**Illustrative**
https://travis-ci.com/helmholtz-analytics/heat/builds/135270829

**Version Info**
Topic branch, but master would suffer from a rebuild.

**Additional comments**
The fix will be to pin the NetCDF dependency to <=1.5.2. Problems start to occur with 1.5.3.

--- a/setup.py
+++ b/setup.py
@@ -35,7 +35,7 @@ setup(
     install_requires=["mpi4py>=3.0.0", "numpy>=1.13.0", "torch==1.3.0"],
     extras_require={
         "hdf5": ["h5py>=2.8.0"],
-        "netcdf": ["netCDF4>=1.4.0"],
+        "netcdf": ["netCDF4>=1.4.0,<=1.5.2"],
         "dev": ["pre-commit>=1.18.3"],
     },
 )

